### DISCUSSION TITLE
CI strategy

### DISCUSSION THREAD

**Discussion by LilianBoulard** [skrub-data/skrub#390] [General]
Hello!

Following a discussion with @Vincent-Maladiere regarding #365, we have a few points we'd like to debate regarding the CI strategy.

First, we seem to agree that it might not be useful to test on MacOS, since the library is written in pure Python, and therefore shouldn't be affected by hardware too much.

Second, we wondered how many versions we should test per package. For now, we settled on 2 (the minimum and latest versions), but we think that 3 might be better (minimum, "middle", and latest versions).

Please let us know what you think !

**Comment by GaelVaroquaux** [comment]
> First, we seem to agree that it might not be useful to test on MacOS, since the library is written in pure Python, and therefore shouldn't be affected by hardware too much.

Not convinced. Weird things happen.

> Second, we wondered how many versions we should test per package. For now, we settled on 2 (the minimum and latest versions), but we think that 3 might be better (minimum, "middle", and latest versions).

Let's start with 2, and add version combinations as we find weird / failing behaviors (which we will).

**Comment by Vincent-Maladiere** [comment]
Hello! I have just finished drafting [the PR](https://github.com/dirty-cat/dirty_cat/pull/365) and it works like a charm :)
